Configure changes to build OpenPBS with Python 3.10

toonen opened the pull request:
This patch updates the autoconf PBS_AC_WITH_PYTHON macro to include support for Python 3.10.  It also removes support for Python 3.5, which reached end-of-life over a year ago and lacks support for f-strings.

bhroam reviewed (commented):
I'm not really an expert on M4, but I do have a couple of questions.  First, you have a space between '-' and 'o'.  I suspect that should be '-o' for OR?  Second, previously we added --embed for 3.9.  You removed that.  Is that not needed for 3.9 any more?  (3.10?).

toonen commented:
@bhroam Thanks for pointing out the space in `- o`.  That was a typo that went undetected because of the way the m4/sh code was structured.  In 7ec4a3d2a7e26069fda5a9b88075e9b387c3ab5a, I reversed the logic and restructured the shell code so that typos in the conditional will be considered a failure.  I also changed the logic to allow any Python 3 version where the minor version number was 6 or greater.  The idea being that we'll fix it when a build against Python 3.X, X greater than 10, breaks rather than fixing it every time a new minor version of Python 3 is released.

As for the logic to include the `--embed` flag, I had changed that to be `$python_minor_version -ge 8` so that it would be added for version 3.8 and beyond.  Again, we can change the logic later if necessitated by a future Python 3 release.

bhroam reviewed (commented):
I like the new format of the python if statement, but I liked it before when you could only have 3.6-3.10.  Without this, we still can't use f strings because we might be using 3.5 still.

bayucan reviewed (approved):
Looks good.

bayucan commented:
Merged #2510 into master.
